.github/workflows/test.yaml

```
@@ -93,6 +93,31 @@ jobs:
           filter-mode: nofilter
           workdir: ${{ inputs.working-directory }}
           woke-version: latest
+  vale:
+    name: Style checker
+    runs-on: >-
+      ${{
+        inputs.self-hosted-runner &&
+        fromJson(format('[''self-hosted'', ''{0}'', ''{1}'', ''{2}'']',
+          inputs.self-hosted-runner-arch, inputs.self-hosted-runner-label, inputs.self-hosted-runner-image
+        )) || 'ubuntu-22.04'
+      }}
+    defaults:
+      run:
+        shell: bash
+        working-directory: ${{ inputs.working-directory }}
+    steps:
+      - name: Checkout repo to runner
+        uses: actions/checkout@v4.1.7
+      - run: |
+          sudo apt-get update -y && sudo apt-get install -y tree ruby-rubygems
+      - name: Install styles
+        uses: canonical/praecepta@main
+      - name: Run Vale tests
+        uses: errata-ai/vale-action@reviewdog
+        with:
+          files: ./docs
```

```suggestion
          files: ${{ inputs.charm-directory }}/docs
```

This would allow the check to run for paas-app-charmer charms, which usually have the docs living in `charm/docs`, see https://github.com/canonical/netbox/tree/main/charm/docs and https://github.com/canonical/netbox/blob/main/.github/workflows/test.yaml#L11 .